lib/auth/auth.go

```diff
@@ -841,63 +842,52 @@ func (a *Server) PreAuthenticatedSignIn(user string, identity tlsca.Identity) (s
 	return sess.WithoutSecrets(), nil
 }
 
-// U2FAuthenticateChallenge is a U2F authentication challenge sent on user
+// MFAAuthenticateChallenge is a U2F authentication challenge sent on user
 // login.
-type U2FAuthenticateChallenge struct {
+type MFAAuthenticateChallenge struct {
 	// Before 6.0 teleport would only send 1 U2F challenge. Embed the old
 	// challenge for compatibility with older clients. All new clients should
 	// ignore this and read Challenges instead.
 	*u2f.AuthenticateChallenge
+
 	// The list of U2F challenges, one for each registered device.
-	Challenges []u2f.AuthenticateChallenge `json:"challenges"`
+	U2FChallenges []u2f.AuthenticateChallenge `json:"u2f_challenges"`
+	// TOTPChallenge specifies whether TOTP is supported for this user.
+	TOTPChallenge bool
```

Comment:
Does this field get marshaled? There's no json tag.

Reply:
added JSON tag